src/main/java/org/cryptomator/ui/error/ErrorController.java

```diff
@@ -95,13 +127,141 @@ public void copyDetails() {
+
+				if (value.isPresent()) {
+					matchingErrorDiscussion = value.get();
+					errorSolutionFound.set(true);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Checks if an ErrorDiscussion object is a partial match based on the presence of the error code's method code in its title.
+	 *
+	 * @param errorDiscussion The ErrorDiscussion object to be checked.
+	 * @return A boolean value indicating if the ErrorDiscussion object is a partial match:
+	 * - true if the object's title contains the error code's method code,
+	 * - false otherwise.
+	 */
+	public boolean isPartialMatchFilter(ErrorDiscussion errorDiscussion) {
+		return errorDiscussion.title.contains(" " + errorCode.methodCode());
+	}
+
+	/**
+	 * Compares two ErrorDiscussion objects based on their upvote counts and returns the result.
+	 *
+	 * @param ed1 The first ErrorDiscussion object.
+	 * @param ed2 The second ErrorDiscussion object.
+	 * @return An integer indicating which ErrorDiscussion object has a higher upvote count:
+	 * - A positive value if ed2 has a higher upvote count than ed1,
+	 * - A negative value if ed1 has a higher upvote count than ed2,
+	 * - Or 0 if both upvote counts are equal.
+	 */
+	public int compareUpvoteCount(ErrorDiscussion ed1, ErrorDiscussion ed2) {
+		return Integer.compare(ed2.upvoteCount, ed1.upvoteCount);
+	}
+
+	/**
+	 * Compares two ErrorDiscussion objects based on their answered status and returns the result.
+	 *
+	 * @param ed1 The first ErrorDiscussion object.
+	 * @param ed2 The second ErrorDiscussion object.
+	 * @return An integer indicating the answered status of the ErrorDiscussion objects:
+	 * - A negative value (-1) if ed1 is considered answered and ed2 is considered unanswered,
+	 * - A positive value (1) if ed1 is considered unanswered and ed2 is considered answered,
+	 * - Or 0 if both ErrorDiscussion objects are considered answered or unanswered.
+	 */
+	public int compareIsAnswered(ErrorDiscussion ed1, ErrorDiscussion ed2) {
+		if (ed1.answer != null && ed2.answer == null) {
+			return -1;
+		} else if (ed1.answer == null && ed2.answer != null) {
+			return 1;
+		} else {
+			return 0;
+		}
+	}
+
+	/**
+	 * Compares two ErrorDiscussion objects based on the presence of an exact match with the error code in their titles and returns the result.
+	 *
+	 * @param ed1 The first ErrorDiscussion object.
+	 * @param ed2 The second ErrorDiscussion object.
+	 * @return An integer indicating the comparison result based on the presence of an exact match with the error code in the titles:
+	 * - A negative value (-1) if ed1 has an exact match with the error code in the title and ed2 does not have a match,
+	 * - A positive value (1) if ed1 does not have a match and ed2 has an exact match with the error code in the title,
+	 * - Or 0 if both ErrorDiscussion objects either have an exact match or do not have a match with the error code in the titles.
+	 */
+	public int compareExactMatch(ErrorDiscussion ed1, ErrorDiscussion ed2) {
```

Comment:
my name suggestion:

```suggestion
	public int compareByFullErrorCode(ErrorDiscussion ed1, ErrorDiscussion ed2) {
```

Reply:
(don't forget to also update the names of the test methods)

Reply:
method and test have been renamed